src/visit_mut.rs

```diff
@@ -1,25 +1,93 @@
 use syn::visit_mut::VisitMut;
 use syn::{
-    Arm, Expr, ExprAssign, ExprBinary, ExprCall, ExprClosure, ExprField, ExprForLoop, ExprIf,
-    ExprLet, ExprLoop, ExprMatch, ExprMethodCall, ExprPath, ExprUnary, ExprWhile, Field, Fields,
-    FnArg, ItemConst, ItemEnum, ItemFn, ItemImpl, ItemStatic, ItemStruct, ItemTrait, ItemType,
-    ItemUnion, Macro, Member, Pat, PatIdent, PatTuple, PatType, Path, PathSegment, ReturnType,
-    Signature, Token, Type, Variant,
+    Arm, ConstParam, Expr, ExprAssign, ExprBinary, ExprCall, ExprClosure, ExprField, ExprForLoop,
+    ExprIf, ExprLet, ExprLoop, ExprMatch, ExprMethodCall, ExprPath, ExprUnary, ExprWhile, Field,
+    Fields, FnArg, ImplItem, Item, ItemConst, ItemEnum, ItemFn, ItemImpl, ItemStatic, ItemStruct,
+    ItemTrait, ItemType, ItemUnion, Macro, Member, Pat, PatIdent, PatTuple, PatType, Path,
+    PathSegment, ReturnType, Signature, Token, Type, TypeParam, Variant,
 };
 
 use crate::ident_visitor::IdentVisitor;
 
+fn get_ident(item: &Item) -> Option<syn::Ident> {
+    match item {
+        Item::Const(item_const) => Some(item_const.ident.clone()),
+        Item::Enum(item_enum) => Some(item_enum.ident.clone()),
+        Item::Fn(item_fn) => Some(item_fn.sig.ident.clone()),
+        Item::ExternCrate(item_extern_crate) => Some(item_extern_crate.ident.clone()),
+        Item::Impl(_) => None,
+        Item::ForeignMod(item_mod) => item_mod
+            .abi
+            .name
+            .clone()
+            .map(|name| Ident::new(&name.value(), name.span())),
+        Item::Macro(item_macro) => item_macro.ident.clone(),
+        Item::Static(item_static) => Some(item_static.ident.clone()),
+        Item::Struct(item_struct) => Some(item_struct.ident.clone()),
+        Item::Trait(item_trait) => Some(item_trait.ident.clone()),
+        Item::TraitAlias(item_trait_alias) => Some(item_trait_alias.ident.clone()),
+        Item::Type(item_type) => Some(item_type.ident.clone()),
+        Item::Union(item_union) => Some(item_union.ident.clone()),
+        Item::Use(_) => None,
+        Item::Verbatim(_) => None,
+        _ => None,
+    }
+}
+
+fn get_impl_item_ident(item: &ImplItem) -> Option<syn::Ident> {
+    match item {
+        ImplItem::Const(item_const) => Some(item_const.ident.clone()),
+        ImplItem::Fn(item_method) => Some(item_method.sig.ident.clone()),
+        ImplItem::Macro(_) => None,
+        ImplItem::Type(item_type) => Some(item_type.ident.clone()),
+        ImplItem::Verbatim(_) => None,
+        _ => None,
+    }
+}
+
 impl VisitMut for IdentVisitor {
+    fn visit_file_mut(&mut self, i: &mut syn::File) {
+        i.items
+            .sort_by(|item1, item2| match (get_ident(item1), get_ident(item2)) {
+                (Some(ident1), Some(ident2)) => ident1.cmp(&ident2),
+                (Some(_), None) => std::cmp::Ordering::Less,
+                (None, Some(_)) => std::cmp::Ordering::Greater,
+                (None, None) => std::cmp::Ordering::Equal,
+            });
```

Comment:
Is it possible to use standard ordering here as well?

Reply:
No I tried but it didn't work

Reply:
Why not? Obviously the golden tests will have to be adjusted, because the sorting order changes. But is there any other reason why standard sorting order doesn't work for us and we must use a non-standard one?

Reply:
addressed in a9c6c8913293d72eb8822f186d77b94cc33d8cf2